samples/commons/src/main/java/io/javaoperatorsdk/webhook/sample/commons/AdmissionControllers.java

```diff
@@ -19,46 +19,22 @@ public class AdmissionControllers {
 
   // adds a label to the target resource
   public static AdmissionController<Ingress> mutatingController() {
-    return new AdmissionController<>((resource, operation) -> {
-      if (resource.getMetadata().getLabels() == null) {
-        resource.getMetadata().setLabels(new HashMap<>());
-      }
-      resource.getMetadata().getLabels().putIfAbsent(MUTATION_TARGET_LABEL, "mutation-test");
-      return resource;
-    });
+    return new AdmissionController<>(new IngressMutator());
   }
 
   // validates if a resource contains the target label
   public static AdmissionController<Ingress> validatingController() {
-    return new AdmissionController<>((resource, operation) -> {
-      if (resource.getMetadata().getLabels() == null
-          || resource.getMetadata().getLabels().get(VALIDATION_TARGET_LABEL) == null) {
-        throw new NotAllowedException("Missing label: " + VALIDATION_TARGET_LABEL);
-      }
-    });
+    return new AdmissionController<>(new IngressValidator());
   }
 
   public static AsyncAdmissionController<Ingress> asyncMutatingController() {
-    return new AsyncAdmissionController<>(
-        (AsyncMutator<Ingress>) (resource, operation) -> CompletableFuture.supplyAsync(() -> {
-          if (resource.getMetadata().getLabels() == null) {
-            resource.getMetadata().setLabels(new HashMap<>());
-          }
-          resource.getMetadata().getLabels().putIfAbsent(MUTATION_TARGET_LABEL, "mutation-test");
-          return resource;
-        }));
+    return new AsyncAdmissionController<>(new IngressMutator());
```

Comment:
This demonstrates the usage of the new convenience constructor. In general this shows how you can de-deduplicate the code for mutators and validators by using old-school inner classes that can be used for sync and async endpoints.

Reply:
If you simply run `mvn clean install`, that should fix it (if remember correctly)

Reply:
Sadly not, `mvn clean install` is successful and no files are changed. The Quarkus/JOSDK repos with spotless worked indeed fine for me.

Reply:
Ahh, sry, for me neither. Would you like to do the migration to spotless or should I do it (not sure in what timeline) ?

Reply:
I'm not a Maven guy (anymore), but I can surely have a look. I'd say this should go in to a new PR and then I rebase this one later?

Reply:
yes, that would be ideal, thx!

Reply:
I've re-created the branch from `main`, cherry-picked each commit and ran `spotless:apply` for all of them. So all commits should be compliant with the code formatting now.